File: gitdot-api/src/endpoint/migration/github/get_github_app_install_url.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::migration::GitHubAppInstallUrlResource};

pub struct GetGitHubAppInstallUrl;

impl Endpoint for GetGitHubAppInstallUrl {
    const PATH: &'static str = "/migration/github/install-url";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetGitHubAppInstallUrlRequest;
    type Response = GetGitHubAppInstallUrlResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetGitHubAppInstallUrlRequest {
    pub action: String,
}

pub type GetGitHubAppInstallUrlResponse = GitHubAppInstallUrlResource;

File: gitdot-web/packages/gitdot-api/src/endpoint/migration/github/get-github-app-install-url.ts
import { z } from "zod";
import { GitHubAppInstallUrlResource } from "../../../resource";
import type { Endpoint } from "../../endpoint";

export const GetGitHubAppInstallUrlRequest = z.object({
  action: z.enum(["migration", "onboarding"]),
});
export type GetGitHubAppInstallUrlRequest = z.infer<
  typeof GetGitHubAppInstallUrlRequest
>;

export const GetGitHubAppInstallUrlResponse = GitHubAppInstallUrlResource;
export type GetGitHubAppInstallUrlResponse = z.infer<
  typeof GetGitHubAppInstallUrlResponse
>;

export const GetGitHubAppInstallUrl = {
  path: "/migration/github/install-url",
  method: "GET",
  request: GetGitHubAppInstallUrlRequest,
  response: GetGitHubAppInstallUrlResponse,
} as const satisfies Endpoint;
export type GetGitHubAppInstallUrl = typeof GetGitHubAppInstallUrl;
